lib/dry/view/decorator.rb

```diff
@@ -27,6 +28,15 @@ def call(name, value, renderer:, context:, **options)
       def part_class(name, value, **options)
         options.fetch(:as) { Part }
       end
+
+      private
+
+      # @api private
+      def singularize_options(**options)
+        options = options.dup
```

Comment:
TIL! Thanks @flash-gordon, I'll adjust this now and I'll internalise this knowledge for the future :)